colin-api/src/colin_api/models/business.py

```diff
@@ -671,3 +671,93 @@ def reset_corp_states(cls, cursor, event_ids: List):
+        business = None
+        try:
+            # get record
+            if not con:
+                con = DB.connection
+                # con.begin()
+
+            cursor = con.cursor()
+            cursor.execute(
+                """
+                select corp.corp_num, corp_typ_cd, recognition_dts, bn_15, can_jur_typ_cd, othr_juris_desc,
+                    filing.period_end_dt, last_agm_date, corp_op_state.full_desc as state, admin_email,
+                    corp_state.state_typ_cd as corp_state, corp_op_state.op_state_typ_cd as corp_state_class
+                from CORPORATION corp
+                    join CORP_STATE on CORP_STATE.corp_num = corp.corp_num and CORP_STATE.end_event_id is null
+                    join CORP_OP_STATE on CORP_OP_STATE.state_typ_cd = CORP_STATE.state_typ_cd
+                    left join JURISDICTION on JURISDICTION.corp_num = corp.corp_num
+                    join event on corp.corp_num = event.corp_num
+                    left join filing on event.event_id = filing.event_id and filing.filing_typ_cd in ('OTANN', 'ANNBC')
+                where corp.CORP_NUM=:corp_num
+                order by filing.period_end_dt desc nulls last
+                """,
+                corp_num=corp_num
+            )
+            business = cursor.fetchone()
+            if not business:
+                raise BusinessNotFoundException(corp_num=corp_num)
+
+            # add column names to resultset to build out correct json structure and make manipulation below more robust
+            # (better than column numbers)
+            business = dict(zip([x[0].lower() for x in cursor.description], business))
+            # get all assumed, numbered/corporation, translation names
+            corp_names = CorpName.get_current(cursor=cursor, corp_num=corp_num)
+            assumed_name = None
+            corp_name = None
+            for name_obj in corp_names:
+                if name_obj.type_code == CorpName.TypeCodes.ASSUMED.value:
+                    assumed_name = name_obj.corp_name
+                    break
+                elif name_obj.type_code in [CorpName.TypeCodes.CORP.value, CorpName.TypeCodes.NUMBERED_CORP.value]:
+                    corp_name = name_obj.corp_name
+
+            # get last ledger date from EVENT table and add to business record
+            # note - FILE event type is correct for new filings; CONVOTHER is for events/filings pulled over from COBRS
+            cursor.execute(
+                """
+                select max(EVENT_TIMESTMP) from EVENT
+                where EVENT_TYP_CD in ('FILE', 'CONVOTHER') and CORP_NUM=:corp_num
+                """,
+                corp_num=corp_num
+            )
+            last_ledger_timestamp = cursor.fetchone()[0]
+            business['last_ledger_timestamp'] = last_ledger_timestamp
+            # if this is an XPRO, get correct jurisdiction; otherwise, it's BC
+            if business['corp_typ_cd'] == 'XCP':
```

Comment:
this needs to be updated to include a check for `corp_typ_cd` == `A`.

Reply:
your latest change replaces the `XCP` check which is incorrect.  The `if` clause should be `if business['corp_typ_d'] in ('XCP', 'A')`

Reply:
Can you let me know why we do like that?
I don't know well about `if business['corp_typ_d'] in ('XCP', 'A')`

Reply:
It's because these are corp type codes for extra provincial companies.  Extraprovincial companies make use of the `jurisdiction` table to store info about its home jurisdiction.